kratos/utilities/model_part_graph_utilities.h

```diff
@@ -0,0 +1,480 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:   Riccardo Rossi
+//
+
+#if !defined(KRATOS_MODELPART_GRAPH_UTILITIES_H_INCLUDED)
+#define  KRATOS_MODELPART_GRAPH_UTILITIES_H_INCLUDED
+
+
+// System includes
+#include <string>
+#include <iostream>
+#include <queue>
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "containers/sparse_contiguous_row_graph.h"
+#include "utilities/parallel_utilities.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This file computes the graph representing the connectivity of a modelpart. 
+/** Given a modelpart, it returns the csr_representation of its graph.
+*/
+class ModelPartGraphUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ModelPartGraphUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(ModelPartGraphUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    ModelPartGraphUtilities(){}
+
+    /// Destructor.
+    virtual ~ModelPartGraphUtilities(){}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**The following function computes the connectivity graph (based on node.Id()-1 so that ids start in 0) for the mesh used as input.
+    * @param rModelPart modelpart of which we will compute the graph
+    * @retval the Graph being computed
+    *
+    * NOTE: this function is suboptimal if discontinuous or very large ids are employed 
+    (the graph will have empty rows from 0 to the id of the largest node)
+    */
+    SparseContiguousRowGraph<> ComputeGraph(const ModelPart& rModelPart){
```

Comment:
If there is no template move the code to a cpp please, compilation time is already too much

Reply:
done